docs/proposals/default-service-plans.md

```diff
@@ -0,0 +1,544 @@
+
+
+### Service Class
+
+
+| Field | Description |
+|-------|-------------|
+| Service Type | Examples: mysql, redis|
+| <p>Default Provision Parameters</p><p>Default Bind Parameters</p> | <p>Default parameters to supply to the broker during provisioning and binding.</p> <p>OSB API V3 - Make the schema required, and defaults optional.|
+| Default Bind Response Transform | <p>Default json key mapping to transform a non-standard broker response into something that can be relied upon.</p> <p>For example, if a broker returns "database.name" (a nested value) in the response, it can be changed to "database" to un-nest it and apply a standard name.</p>|
+| Controller Reference |<p>Set by Service Catalog when populating the catalog from a broker list to indicate that it is managed by the controller.</p> <p>When not present, the class is ignored during a relist. This gives operators a way to define custom classes.</p>|
+
+Notes
+
+* Relist needs to take into account if the class is managed by the controller or was created by the cluster operator.
+* Defaults can be defined at the Cluster or Namespace level.
+* For simplification, only broker bind response bodies can be transformed, not provision, since that's where we require that functionality today.
+
+
+### Service Plan
+
+| Field | Description |
+|-------|-------------|
+| Service Type | Examples: mysql, redis |
+| annotation: is-default-plan=true | <p>Specifies whether or not a Service Plan is the default for its Service Type.</p> <p>When annotated as the default and a Service Type is set, this plan is used to dynamically provision an instance for that type.</p> |
+| Controller Reference|<p>Set by Service Catalog when populating the catalog from a broker list to indicate that it is managed by the controller.</p> <p>When not present, the plan is ignored during a relist. This gives operators a way to define custom plans.</p> |
+
+
+Notes:
+
+* The controller will enforce that only one default is set per service type.
+* For simplification, defaults are set at the class level only, plans cannot override them.
+
+
+### Service Instance
+
+
+| Field | Description |
+|-------|-------------|
+| Service Type | Examples: mysql, redis |
+| Provision Parameters | Overrides the default provision parameters defined on the class/plan. |
+
+Notes:
+* Merge provision defaults and overrides before sending to the broker.
+
+
+### Service Binding
+
+| Field | Description |
+|-------|-------------|
+| Service Type | Examples: mysql, redis |
+| Service Plan | When a ServicePlan is not specified, the default ServicePlan for the type is selected. |
+| Bind Parameters|Overrides the default bind parameters defined on the class/plan. |
+| Bind Response Transform|Allow overriding the default bind response transform set at the class level. |
```

Comment:
I think this one is being worked on already or there is part of an impl in the repo already

Reply:
Thanks for the heads up! I'll update the proposal to use the same terms as what is being added right now.